pkg/templates/nstemplatetiers/nstemplatetier_generator_whitebox_test.go

```diff
@@ -8,191 +8,235 @@ import (
 	texttemplate "text/template"
 
 	toolchainv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/host-operator/pkg/apis"
 	testnstemplatetiers "github.com/codeready-toolchain/host-operator/test/templates/nstemplatetiers"
 
-	"github.com/codeready-toolchain/host-operator/pkg/apis"
 	templatev1 "github.com/openshift/api/template/v1"
-	"github.com/pkg/errors"
-	"github.com/satori/go.uuid"
+	uuid "github.com/satori/go.uuid"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/serializer"
 	"k8s.io/client-go/kubernetes/scheme"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
 )
 
-func TestNewNSTemplateTierGenerator(t *testing.T) {
+func TestLoadTemplatesByTiers(t *testing.T) {
 
-	s := scheme.Scheme
-	err := apis.AddToScheme(s)
-	require.NoError(t, err)
+	logf.SetLogger(zap.Logger(true))
 
 	t.Run("ok", func(t *testing.T) {
-		// when initializing the generator with the production Asset
-		g, err := newNSTemplateTierGenerator(s, Asset)
-		// then
-		require.NoError(t, err)
-		// verify that there are tiers with revisions (whatever their values)
-		require.NotEmpty(t, g.revisions)
-		for tier := range g.revisions {
-			assert.NotEmpty(t, g.revisions[tier])
-		}
-	})
 
-	t.Run("failures", func(t *testing.T) {
-
-		t.Run("asset error", func(t *testing.T) {
+		t.Run("with prod assets", func(t *testing.T) {
 			// given
-			asset := func(name string) ([]byte, error) {
-				return nil, errors.New("test")
+			assets := NewAssets(AssetNames, Asset)
+			// when
+			tmpls, err := loadTemplatesByTiers(assets)
+			// then
+			require.NoError(t, err)
+			// then
+			require.NoError(t, err)
+			require.Len(t, tmpls, 3)
+			require.NotContains(t, "foo", tmpls) // make sure that the `foo: bar` entry was ignored
+			for _, tier := range []string{"advanced", "basic", "team"} {
+				t.Run(tier, func(t *testing.T) {
+					for _, kind := range []string{"code", "dev", "stage"} {
+						t.Run(kind, func(t *testing.T) {
+							if tier == "team" && kind == "code" {
+								// not applicable
+								return
+							}
+							assert.NotEmpty(t, tmpls[tier].namespaceTemplates[kind].revision)
+							assert.NotEmpty(t, tmpls[tier].namespaceTemplates[kind].content)
+						})
+						t.Run("cluster", func(t *testing.T) {
```

Comment:
give the context about the kind the cluster is tested for:
```suggestion
						t.Run("cluster " + kind, func(t *testing.T) {
```

Reply:
actually, the `t.Run("cluster", ...)` should not be in the loop that ranges on the namespace kinds 🤦‍♂

Reply:
refactored in 98dca2f

Reply:
ah right :+1: